### ISSUE TITLE
FFI is missing the `clear` method

### ISSUE THREAD

**Issue by fniephaus** [oracle/truffleruby#1687]
[HTMLProofer](https://github.com/gjtorikian/html-proofer) installs fine, but fails with the following error in the `ffi` module:

```ruby
irb(main):001:0> require 'html-proofer'
=> true
irb(main):002:0> HTMLProofer.check_directory('/my/website').run
Running ["ScriptCheck", "ImageCheck", "LinkCheck"] on ["/my/website"] on *.html...


Checking 10 external links...
Traceback (most recent call last):
       13: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/bin/irb:31:in `<main>'
       12: from (irb):2
       11: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/html-proofer-3.10.2/lib/html-proofer/runner.rb:40:in `run'
       10: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/html-proofer-3.10.2/lib/html-proofer/runner.rb:80:in `check_files'
        9: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/html-proofer-3.10.2/lib/html-proofer/runner.rb:117:in `validate_urls'
        8: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/html-proofer-3.10.2/lib/html-proofer/url_validator.rb:29:in `run'
        7: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/html-proofer-3.10.2/lib/html-proofer/url_validator.rb:102:in `external_link_checker'
        6: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/typhoeus-1.3.1/lib/typhoeus/hydra/memoizable.rb:51:in `run'
        5: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/typhoeus-1.3.1/lib/typhoeus/hydra/runnable.rb:15:in `run'
        4: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/ethon-0.12.0/lib/ethon/multi/operations.rb:46:in `perform'
        3: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/ethon-0.12.0/lib/ethon/multi/operations.rb:104:in `reset_fds'
        2: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.6.0/gems/ethon-0.12.0/lib/ethon/curls/classes.rb:39:in `clear'
        1: from graalvm-ce-1.0.0-rc16/Contents/Home/jre/languages/ruby/lib/truffle/ffi/struct.rb:139:in `clear'
NoMethodError (undefined method `clear' for #<Truffle::FFI::MemoryPointer address=0x7fa1f3822600>:Truffle::FFI::MemoryPointer)
```

**Comment by nirvdrum** [comment]
A simpler example that doesn't require using `html-proofer`:

```
require 'ffi'

class SimpleStruct < FFI::Struct
  layout  :value, :long
end

a = SimpleStruct.new
a[:value] = 32
a.clear

puts a[:value]
```

**Comment by fniephaus** [comment]
Thanks for debugging, @nirvdrum!

**Comment by eregon** [comment]
I implemented the FFI::Pointer#clear method in 2216955301616fb2870eb7a479564376453ad597, so the example from @nirvdrum now works.
This will be in the next release.
@fniephaus  Thank you for the report. Please tell us if there are other issues with `html-proofer`.

**Comment by fniephaus** [comment]
Sure, will do! Thanks for the fix :)
